website/docs/r/iothub_file_upload.html.markdown

```diff
@@ -0,0 +1,123 @@
+
+## Example Usage
+
+```hcl
+resource "azurerm_resource_group" "test" {
+  name     = "example-resources"
+  location = "West Europe"
+}
+
+resource "azurerm_storage_account" "test" {
+  name                     = "examplestorage"
+  resource_group_name      = azurerm_resource_group.test.name
+  location                 = azurerm_resource_group.test.location
+  account_tier             = "Standard"
+  account_replication_type = "LRS"
+}
+
+resource "azurerm_storage_container" "test" {
+  name                  = "examplecontainer"
+  storage_account_name  = azurerm_storage_account.test.name
+  container_access_type = "private"
+}
+
+resource "azurerm_iothub" "test" {
+  name                = "example-iothub"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+
+  sku {
+    name     = "S1"
+    capacity = "1"
+  }
+
+  identity {
+    type = "SystemAssigned"
+  }
+}
+
+resource "azurerm_role_assignment" "test_storage_blob_data_contrib" {
+  role_definition_name = "Storage Blob Data Contributor"
+  scope                = azurerm_storage_account.test.id
+  principal_id         = azurerm_iothub.test.identity[0].principal_id
+}
+
+resource "azurerm_iothub_file_upload" "test" {
+  iothub_id         = azurerm_iothub.test.id
+  container_name    = azurerm_storage_container.test.name
+  connection_string = azurerm_storage_account.test.primary_blob_connection_string
+
+  authentication_type = "identityBased"
+
+  depends_on = [
+    azurerm_role_assignment.test_storage_blob_data_contrib,
+  ]
+}
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `connection_string` - (Required) The connection string for the Azure Storage account to which files are uploaded.
+
+* `container_name` - (Required) The name of the root container where you upload files. The container need not exist but should be creatable using the connection_string specified.
+
+* `iothub_id` - (Required) The ID of the IoTHub. Changing this forces a new IotHub File Upload to be created.
+
+---
+
+* `authentication_type` - (Optional) Type used to authenticate against the storage account. Possible values are `keyBased` and `identityBased`. Defaults to `keyBased`.
+
+* `default_ttl` - (Optional) The period of time for which a file upload notification message is available to consume before it is expired by the IoT hub, specified as an [ISO 8601 timespan duration](https://en.wikipedia.org/wiki/ISO_8601#Durations). This value must be between 1 minute and 48 hours, and evaluates to `PT1H` by default.
+
+* `identity_id` - (Optional) ID of the User Managed Identity used to authenticate against the storage account.
+
+-> **NOTE:** `identity_id` can only be specified when `authentication_type` is `identityBased`. It must be one of the `identity_ids` of the Iot Hub. If not specified when `authentication_type` is `identityBased`, System Assigned Managed Identity of the Iot Hub will be used.
```

Comment:
```suggestion
-> **NOTE:** `identity_id` can only be specified when `authentication_type` is `identityBased`. It must be one of the `identity_ids` of the Iot Hub. If `identity_id`is omitted when `authentication_type` is `identityBased`, then the System Assigned Managed Identity of the Iot Hub will be used.
```
Same comment here, does the `azurerm_iothub` resource have the plural property `identity_ids`?